pkg/skaffold/schema/profiles.go

```diff
@@ -55,11 +59,36 @@ func ApplyProfiles(c *latest.SkaffoldConfig, opts cfg.SkaffoldOptions) error {
 		}
 	}
 
+	return enableEffectiveKubecontext(isContextSpecific, opts.KubeContext, c.Deploy.KubeContext)
+}
+
+func enableEffectiveKubecontext(isContextImmutable bool, cliContext, effectiveContext string) error {
+	// cli flag takes precedence
+	if cliContext != "" {
+		return nil
+	}
+
+	kubeConfig, err := kubectx.CurrentConfig()
```

Comment:
why do we need to retrieve this if we have the `cliContext` set already?

Reply:
This is tricky. There are (will be) four sources for the kube-context:
* the CLI, which is passed in as `cliContext`
* the global config, which will modify `kubeConfig.CurrentContext`
* the default value of `kubeConfig.CurrentContext` (i.e. the kubeconfig setting)
* the value from `skaffold.yaml`

In essence, we need to know at this point whether the CLI flag was given, because it should always take precedence. As currently implemented, the CLI flag would also not produce an error if some kube-context specific profile was activated.

Reply:
I see, so, it is mainly inspired by the global config part right? 
That might take a while to implement, and I'm starting to think that we might be a bit overengineering with it. What if it will never exist? Would we simplify this line?

Reply:
So if we leave the global config out of the picture for now, I think it will be possible to call `UseKubeContext` a single time (and immediately lock any further mutations of the kubecontext).

However, in `profile.go` we would still have to call out to `kubectx.CurrentConfig` here, because we need to know the default kubecontext (from the kubeconfig).